### PR TITLE
Revert node requirement to >=10

### PR DESCRIPTION
**Pull request by ankhyk**
- @apidevtools/swagger-parser has problem supporting node <=14(with npm < 7): [swagger-parser 10.1.0 is broken when used with npm < 7 or using --legacy-peer-deps](https://github.com/ajv-validator/ajv-draft-04)
```
`-- @apidevtools/swagger-parser@10.1.0
  `-- ajv-draft-04@1.0.0
```
- @apidevtools/json-schema-ref-parser has fix the compatibility, so we can revert @apidevtools/swagger-parser to swagger-parser

### CODE REVIEW

**Comment by raych1** [comment]
@mikeharder , we have to revert this as it causes runtime issue on node <=14.

File: package.json
Comment (mikeharder) [inline package.json]:
```suggestion
    "@apidevtools/swagger-parser": "10.0.3",
```

`swagger-parser` is just a redirect to `@apidevtools/swagger-parser`, so I think it's better to directly depend on the latter.  The only reason `swagger-parser@^10.0.3` even works, is because the maintainers didn't publish `swagger-parser@10.1.0` when they published `@apidevtools/swagger-parser@10.1.0`.  I'm not sure if this was intentional or a mistake.

If you depend on `swagger-parser@^10.0.3`, and the maintainers do publish a new version of `swagger-parser` in the future, you will be broken again.

File: package.json
Comment (mikeharder) [inline package.json]: What versions of Node do you actually want to guarantee are compatible with `oav`?  Personally, I would use Node 14, since it's the oldest version of Node still supported.

Ideally, you should also run automated tests on all the versions of Node you claim to support.  So if `engines` specifies `node >= 14.0.0`, you should run tests on Node 14, 16, and 18.

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: package.json
Comment (ankhyk) [reply package.json]: Thanks for reminding, I'll lock @apidevtools/swagger-parser to v10.0.3

File: package.json
Comment (raych1) [reply package.json]: Let's keep node 12. Then later on we can retire 12 support with major version upgrade. we should run tests against different node version in CI pipeline.

File: ChangeLog.md
Comment (mikeharder) [inline ChangeLog.md]:
```suggestion
```

**Review by mikeharder — APPROVED** [review]
Approved pending my suggestion

File: package.json
Comment (mikeharder) [reply package.json]: I personally think you can drop support for Node 12 (or any EOL Node version) in either a major or a minor/patch upgrade.

Per strict semver, dropping support for a Node version is probably considered a breaking change.

However, one could also argue:
* Packages should drop support for Node versions when they stop testing on those versions
* It's unreasonable to expect package maintainers to test on Node versions very long after they are EOL
* Packages should not be forced to upgrade their major versions when Node versions go EOL

**Review by mikeharder — APPROVED** [review]

**Comment by raych1** [comment]
@keryul , can you add change log saying 'lock apidevtools/swagger-parser@10.0.3 as 10.1.0 brings conflict of ajv package version'

**Review by raych1 — APPROVED** [review]

**Pull request merged**
